ntoskrnl/mm/ARM3/mdlsup.c

```diff
@@ -1622,29 +1622,297 @@ MmAdvanceMdl(IN PMDL Mdl,
 }
 
 /*
- * @unimplemented
+ * @implemented
  */
 PVOID
 NTAPI
 MmMapLockedPagesWithReservedMapping(IN PVOID MappingAddress,
                                     IN ULONG PoolTag,
-                                    IN PMDL MemoryDescriptorList,
+                                    IN PMDL Mdl,
                                     IN MEMORY_CACHING_TYPE CacheType)
 {
-    UNIMPLEMENTED;
-    return 0;
+    PPFN_NUMBER MdlPages, LastPage;
+    PFN_COUNT PageCount;
+    BOOLEAN IsIoMapping;
+    MI_PFN_CACHE_ATTRIBUTE CacheAttribute;
+    PMMPTE PointerPte;
+    MMPTE TempPte;
+
+    ASSERT(Mdl->ByteCount != 0);
+
+    //
+    // Get the list of pages and count
+    //
```

Comment:
(code style comment: for new code, I would suggest you use the new commenting style, instead of that style)
https://reactos.org/wiki/Coding_Style#Commenting